probe-rs/src/architecture/arm/sequences/mod.rs

```diff
@@ -197,25 +410,23 @@ pub trait ArmDebugSequence: Send + Sync {
     ///
     /// [ARM SVD Debug Description]: http://www.keil.com/pack/doc/cmsis/Pack/html/debug_description.html#debugCoreStart
     #[doc(alias = "DebugCoreStart")]
-    fn debug_core_start(&self, core: &mut Memory) -> Result<(), crate::Error> {
-        use crate::architecture::arm::core::armv7m::Dhcsr;
-
-        let current_dhcsr = Dhcsr(core.read_word_32(Dhcsr::ADDRESS)?);
-
-        // Note: Manual addition for debugging, not part of the original DebugCoreStart function
-        if current_dhcsr.c_debugen() {
-            log::debug!("Core is already in debug mode, no need to enable it again");
-            return Ok(());
+    fn debug_core_start(
+        &self,
+        core: &mut Memory,
+        core_type: CoreType,
+        debug_base: Option<u32>,
+    ) -> Result<(), crate::Error> {
+        // Dispatch based on core type (Cortex-A vs M)
+        match core_type {
+            CoreType::Armv7a => armv7a_core_start(core, debug_base),
+            CoreType::Armv6m | CoreType::Armv7m | CoreType::Armv7em | CoreType::Armv8m => {
+                cortex_m_core_start(core)
+            }
+            _ => panic!(
+                "Logic inconsistency error - non ARM core type passed {:?}",
```

Comment:
I like this error! Tiny nit: maybe we could call this `bug` instead of `error` to make it clear that this is something that should never happen and needs to be fixed and didn't just happen because of false userinput. But maybe that is too nitpicky :D Change at own discretion :)

Reply:
Yeah, I like that - updated